Marlin/src/module/stepper.cpp

```diff
@@ -2509,10 +2509,10 @@ uint32_t Stepper::block_phase_isr() {
       // and steps are repeated twice so dividends have to be scaled and halved
       // and the dividend is directional, i.e. signed
       TERN_(HAS_SHAPING_X, advance_dividend.x = (uint64_t(current_block->steps.x) << 29) / step_event_count);
-      TERN_(HAS_SHAPING_X, if (TEST(current_block->direction_bits, X_AXIS)) advance_dividend.x = -advance_dividend.x);
+      TERN_(HAS_SHAPING_X, if (TEST(current_block->direction_bits, X_AXIS)) advance_dividend.x *= -1);
```

Comment:
Is the compiler smart enough not to do an actual multiplication here?

Reply:
It's a stylistic choice from an old coder. The name of the var appears only once this way instead of twice, making the name a tiny bit easier to edit, if desired.